packages/core/src/schema/schema.test.ts

```diff
@@ -969,6 +969,311 @@ describe('Schema', () => {
+				assert.strictEqual(result.success, false);
+				if (!result.success) {
+					const firstIssue = Array.from(result.issues)[0];
+					assert.ok(firstIssue && firstIssue.message === 'Expected string');
+				}
+			});
+		});
+
+		describe('serialization', () => {
+			const schema = k.enum(['small', 'medium', 'large']);
+
+			it('should serialize values correctly', () => {
+				assert.strictEqual(schema.serialize('small'), 'small');
+				assert.strictEqual(schema.serialize('medium'), 'medium');
+				assert.strictEqual(schema.serialize('large'), 'large');
+			});
+		});
+
+		describe('OpenAPI generation', () => {
+			it('should generate correct OpenAPI schema', () => {
+				const schema = k.enum(['draft', 'published', 'archived']);
+				const openapi = schema.toOpenAPI();
+
+				assert.strictEqual(openapi.type, 'string');
+				assert.deepStrictEqual(openapi.enum, ['draft', 'published', 'archived']);
+			});
+
+			it('should include description and example', () => {
+				const schema = k.enum(['low', 'medium', 'high']).description('Priority level').example('medium');
+				const openapi = schema.toOpenAPI();
+
+				assert.strictEqual(openapi.type, 'string');
+				assert.deepStrictEqual(openapi.enum, ['low', 'medium', 'high']);
+				assert.strictEqual(openapi.description, 'Priority level');
+				assert.strictEqual(openapi.example, 'medium');
+			});
+		});
+
+		describe('single value enum', () => {
+			const schema = k.enum(['only']);
+
+			it('should accept the single valid value', () => {
+				assert.strictEqual(schema.parse('only'), 'only');
+			});
+
+			it('should reject other values', () => {
+				assert.throws(() => schema.parse('other'));
+			});
+		});
+
+		describe('visit method', () => {
+			it('should be a leaf node', () => {
+				const schema = k.enum(['a', 'b', 'c']);
+				assert.doesNotThrow(() => schema.visit());
+			});
+		});
+	});
+
+	describe('KNativeEnum', () => {
+		describe('string enum', () => {
+			enum Color {
+				Red = 'RED',
+				Green = 'GREEN',
+				Blue = 'BLUE',
+			}
+
+			const schema = k.nativeEnum(Color);
+
+			it('should accept valid enum values', () => {
+				assert.strictEqual(schema.parse('RED'), 'RED');
+				assert.strictEqual(schema.parse('GREEN'), 'GREEN');
+				assert.strictEqual(schema.parse('BLUE'), 'BLUE');
+			});
+
+			it('should reject invalid values', () => {
+				assert.throws(() => schema.parse('red'));
+				assert.throws(() => schema.parse('YELLOW'));
+				assert.throws(() => schema.parse(''));
+			});
+
+			it('should reject non-string values when enum is string-based', () => {
+				assert.throws(() => schema.parse(123));
+				assert.throws(() => schema.parse(null));
+				assert.throws(() => schema.parse(undefined));
+			});
+		});
+
+		describe('numeric enum', () => {
+			enum Status {
+				Pending = 0,
+				Active = 1,
+				Inactive = 2,
+			}
+
+			const schema = k.nativeEnum(Status);
+
+			it('should accept valid numeric enum values', () => {
+				assert.strictEqual(schema.parse(0), 0);
```

Comment:
Please also add enum tests for like

assert.strictEqual(schema.parse(Status.Pending), 0);

even though in theory it's still 0, it's nice to assure that contract